subsys/net/ip/net_if.c

```diff
@@ -166,7 +168,17 @@ static bool net_if_tx(struct net_if *iface, struct net_pkt *pkt)
 
 	debug_check_packet(pkt);
 
-	dst = net_pkt_lladdr_dst(pkt);
+	if (!sys_slist_is_empty(&link_callbacks)) {
```

Comment:
Please add comment along the lines of "If there're any link callbacks, with such a callback receiving a destination address, copy that address out of packet, in case packet is freed before callback is called."